debacl/level_set_tree.py

```diff
@@ -1260,6 +1260,72 @@ def construct_tree(X, k, prune_threshold=None, num_levels=None, verbose=False):
     return tree
 
 
+def construct_tree_from_precomputed_matrix(X, p, k, prune_threshold=None, num_levels=None, verbose=False):
+    """
+    Construct a level set tree from tabular data.
+
+    Parameters
+    ----------
+    X : 2-dimensional numpy array
+        A square distance matrix
+
+    p: int
+        Underlying dimensionality of the data that was used to create X, the distance matrix
+
+    k : int
+        Number of observations to consider as neighbors to a given point.
+
+    prune_threshold : int, optional
+        Leaf nodes with fewer than this number of members are recursively
+        merged into larger nodes. If 'None' (the default), then no pruning
+        is performed.
+
+    num_levels : int, optional
+        Number of density levels in the constructed tree. If None (default),
+        `num_levels` is internally set to be the number of rows in `X`.
+
+    verbose : bool, optional
+        If True, a progress indicator is printed at every 100th level of tree
+        construction.
+
+    Returns
+    -------
+    T : LevelSetTree
+        A pruned level set tree.
+
+    See Also
+    --------
+    construct_tree, construct_tree_from_graph, LevelSetTree
+
+    Examples
+    --------
+    >>> X = numpy.random.rand(100, 2)
+    >>> X = scipy.spatial.distance.pdist(data)
```

Comment:
I know this is super pedatic, but I prefer not to overwrite the original data. Maybe call this `D` or `distance_matrix`?